src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2477,3 +2483,72 @@ def _aks_mesh_update(
         return None
 
     return aks_update_decorator.update_mc(mc)
+
+def aks_mesh_get_mesh_config(cmd, client, resource_group_name, name):
+    from azure.cli.core import __version__ as local_version
+    from packaging.version import parse
+    from pprint import pprint
+    from kubernetes.client import CustomObjectsApi
+    
+    kubernetes.config.load_kube_config()
+    v1 = kubernetes.client.CoreV1Api()
+    namespace = 'istio-system'
+    api_response = v1.list_namespaced_config_map(namespace)
+    # print(api_response)
+    namespaces = v1.list_namespace
+    api_items = api_response.items[0]
+    meshconfig = api_items.data['mesh']
+    istio_version = api_items.metadata.labels['operator.istio.io/version']
+    dct = yaml.safe_load(meshconfig)
+    config_set = set()
+    proxy_set = set()
+    version = parse(local_version)
+    
+
+    
+    
+    
+
+    print("AZ version:")
+    print(version)
+    print("-----------------------------------")
+    show_result = aks_show(cmd, client, resource_group_name, name)
+    is_cilium_installed(show_result)
+    is_osm_installed(show_result)
+    print("Kubernetes version:")
+    print(show_result.kubernetes_version)
+    print('----------------------------------')
+
+    
+    for keys in dct.keys():
+        config_set.add(keys)
+    proxy_config = dct['defaultConfig']
+    for keys in proxy_config.keys():
+        proxy_set.add(keys)
+    print("Meshconfig current configuration:")
+    print(config_set)
+    print('----------------------------------')
+    print("Proxyconfig current configuration: ")
+    print(proxy_set)
+    print('------------------------------------')
+    return
+
+def is_osm_installed(show_result):
+    if (show_result.addon_profiles):
+        if (show_result.addon_profiles['openServiceMesh'].enabled == True):
+            print("OSM addon installed.")
+    else:
+        print("OSM addon not installed")
+
+def is_cilium_installed(show_result):
+    if (show_result.network_profile.network_dataplane == "cilium"):
```

Comment:
can use mc here as well:
```suggestion
mc.network_policy == CONST_NETWORK_DATAPLANE_CILIUM
```

Reply:
probably is mc.network_policy.network_dataplane